### ISSUE TITLE
Release German Electra

### ISSUE THREAD

**Issue by Timoeller** [deepset-ai/FARM#567]
release German Electra models in coop with @stefan-it to huggingface modelszoo + PR

Check Electra related issues, see #519

**Comment by stefan-it** [comment]
😍

**Comment by Timoeller** [comment]
We uploaded the next generation of German language models to the modelhub. For our most powerful model see:
https://huggingface.co/deepset/gelectra-large

Still working on the model cards by putting the paper on arxiv and linking it back to the model cards. 

You should be good to test them out yourself now. They are quite powerful :D albeit compute hungry.

**Comment by lucacampanella** [comment]
Hi @Timoeller and thanks again for uploading the electra model.
I'm testing `deepset/gelectra-base` on the database I'm currently working on (news articles). I have some problems to reach the same performance as `deepset/gbert-base`.
Here some comparisons.
The best result obtained with `deepset/gbert-base`:
```
Epochs: 3
Lr: 0.00005 
Eval (dev) macro avg f1: 0.76963
Test macro avg f1: 0.74366
```

With the same hyperameters, but using `deepset/gelectra-base`:
```
Eval macro avg f1: 0.56534
Test macro avg f1: 0.54856
````
0.20 difference!

I noticed the performance is very sensible to the learning rate and the number of epochs. When I run with bigger learning rate and smaller epochs, I get a bad result, while when I decrease the learning rate, but run for longer epochs the results are better.
Here are a few examples of this behaviour (still with `deepset/gelectra-base`:

```
Epochs | Lr      | MAvg F1 eval |  MAvg F1 test 
     3 | 0.00002 |      0.40797 |       0.44812
     4 | 0.00005 |      0.62023 |       0.63105
     8 | 0.00005 |      0.71846 |       0.70248
```
I noticed very similar behaviour with the other german electra model available (`german-nlp-group/electra-base-german-uncased`).

Do you have any insights into this? It's striking to see so much difference between electra and bert and even more to see how sensible electra is to (relatively) small differences in learning rate and epochs.
The code is exactly the same and I've check the right classes for the tokenizer and the model are loaded.
Thanks!

**Comment by brandenchan** [comment]
Hi @lucacampanella, glad to see that you're already using our new GBERT and GELECTRA models! 

Those are some very interesting data points for sure. So to give you our intuition, ELECTRA and BERT models should train at the same rates and be equally sensitive to hyperparameters. After pretraining, their architectures are essentially the same and  GELECTRA and GBERT were also trained using the same data. 

As a first step, I'd suggest we check which version you're using. Are you doing these experiments with Transformers or FARM? If you're using FARM, could you make sure you're using a version that is at least later than #599? This fixes an issue with ELECTRA models at inference time, specifically with the initialisation of the Summary Writer layer.

One thing worth mentioning is that we know GELECTRA is somewhat undertrained (notice the slope in the F1 graph in Figure 1 [here](https://arxiv.org/pdf/2010.10906.pdf)). I'd expect the model to be maybe 1% or 2% worse than GBERT but not 20%. 

If versioning doesn't improve the performance, it would be interesting to see whether GBERT also improves like GELECTRA when trained for more epochs! It would definitely give us a good hint as to what's going wrong!
